src/inject.js

```diff
@@ -44,12 +44,21 @@ const injectStyleTag = (cssContents /* : string */) => {
         }
     }
 
+    const sheet = ((styleTag.styleSheet || styleTag.sheet /* : any */) /* : CSSStyleSheet */);
 
-    if (styleTag.styleSheet) {
-        // $FlowFixMe: legacy Internet Explorer compatibility
-        styleTag.styleSheet.cssText += cssContents;
+    if (sheet.insertRule) {
+        cssContents.forEach((rule) => {
+            try {
+                sheet.insertRule(rule, sheet.cssRules.length);
+            } catch(e) {
+                // The selector for this rule wasn't compatible with the browser
+            }
+        });
     } else {
-        styleTag.appendChild(document.createTextNode(cssContents));
+        const currentStyleTag = styleTag;
+        cssContents.forEach((rule) => {
+            currentStyleTag.innerText = (currentStyleTag.innerText || '') + rule;
+        });
```

Comment:
We probably want to batch update this, like `currentStyleTag.innerText = (currentStyleTag.innerText || '') + cssContents.join('');`.